p4-16/psa/PSA.mdk

```diff
@@ -489,23 +496,35 @@ copy going to the control CPU to have a software-defined header
 (defined in the P4 program) that is different than any headers on the
 packet copies going to the Egress control block.
 
+Whenever the pseudocode above indicates that a packet should be sent
+to a particular place, a PSA implementation may under some
+circumstances instead drop the packet.  For example, the packet buffer
+may be too low on available space for storing new packets, or some
+other congestion control mechanism such as RED (Random Early
+Detection) or AFD (Approximate Fair Dropping) may select the packet
+for dropping.  It is recommended that an implementation maintain
+counters of packets dropped, preferably with separate counters for as
+many different reasons as the implementation has for dropping packets
+outside the control of the P4 program.
```

Comment:
Good to double-check me on this addition.  I believe it must be true, since many traffic patterns must cause queues to build up.

Reply:
It's a fair point -- no actual device has infinite resources to buffer all packets.